src/main/java/com/jarrvis/ticketbooking/ui/controller/ScreeningController.java

```diff
@@ -0,0 +1,95 @@
+package com.jarrvis.ticketbooking.ui.controller;
+
+
+import com.jarrvis.ticketbooking.application.service.ScreeningService;
+import com.jarrvis.ticketbooking.ui.dto.request.AddNewScreeningRequest;
+import com.jarrvis.ticketbooking.ui.dto.request.SearchScreeningRequest;
+import com.jarrvis.ticketbooking.ui.dto.response.ScreeningResource;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import javax.validation.Valid;
+import java.net.URI;
+
+
+/**
+ * REST API endpoint for multiplex screenings related operations
+ */
+@Slf4j
+@RestController
+@RequestMapping(value = "/screenings")
+public class ScreeningController {
+
+    private final ScreeningService screeningService;
+
+    public ScreeningController(
+            final ScreeningService screeningService) {
+        this.screeningService = screeningService;
+    }
+
+    /**
+     * REST operation to add new screening to multiplex offer
+     */
+    @PostMapping()
+    @PreAuthorize("isAuthenticated()")
+    @ApiOperation(value = "Adds new screening to multiplex offer")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Screening added to multiplex offer"),
+            @ApiResponse(code = 401, message = "Missing required authorization"),
+            @ApiResponse(code = 409, message = "Movie does not exist or room does not exist or screening overlaps with existing screenings"),
+            @ApiResponse(code = 422, message = "Add new screening body parameters contains validation errors"),
+    })
+    public Mono<ResponseEntity> add(
+            @ApiParam(value = "Screening details to be added to offer") @RequestBody @Valid AddNewScreeningRequest addNewScreeningRequest,
+            BindingResult bindingResult) throws NoSuchMethodException, MethodArgumentNotValidException {
+
+        log.debug("Add new screening request received: {}", addNewScreeningRequest);
+        if (bindingResult.hasErrors()) {
```

Comment:
czy tej obsługi błędów nie dałoby się ujednolicić i wyrzucić do jednego miejsca?

Reply:
jak najbardziej. właściwie to w moim kodzie można to spokojnie wyrzucić, spring rzuci `MethodArgumentNotValidException` OOTB. to tylko furtka. może się przydać kiedy trzeba rozróżniać miedzy obsługą walidacji na różnych polach.